components/org.wso2.carbon.identity.oauth.endpoint/src/test/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtilTest.java

```diff
@@ -342,10 +356,25 @@ public void testGetUserConsentURL(Object oAuth2ParamObject, boolean isOIDC, bool
                     "is not found in url");
             Assert.assertTrue(ArrayUtils.contains(scopeArray, "internal_login"), "internal_login " +
                     "scope parameter value is not found in url");
+
             if (queryString != null && cacheEntryExists) {
                 Assert.assertTrue(consentUrl.contains(queryString), "spQueryParams value is not found in url");
             }
 
+            if (parameters.getScopes().contains("openid")) {
+                String requestedClaimString = URLDecoder.decode(consentUrl, "UTF-8");
```

Comment:
is it actually the requested claim string that's getting added to the variable `requestedClaimString`? seems like its the whole URL decoded consent URL here. lets name it appropriately.